07-multi-hop/test_spec07.py

```diff
@@ -0,0 +1,240 @@
+import time
+
+import pytest
+
+from riotctrl_shell.gnrc import GNRCICMPv6Echo, GNRCIPv6NIB, GNRCPktbufStats
+from riotctrl_shell.netif import Ifconfig, IfconfigListParser
+
+from testutils.shell import pktbuf, lladdr, ping6, GNRCUDP
+
+
+APP = 'tests/gnrc_udp'
+TO_ADDR = "affe::1"     # address for the first statically routed node
+FROM_ADDR = "beef::1"   # address for the last statically routed node
+pytestmark = pytest.mark.rc_only()
+
+
+class Shell(Ifconfig, GNRCICMPv6Echo, GNRCIPv6NIB, GNRCPktbufStats, GNRCUDP):
+    pass
+
+
+@pytest.fixture
+def statically_routed_nodes(riot_ctrl):
+    nodes = (
+        riot_ctrl(0, APP, Shell),
+        riot_ctrl(1, APP, Shell),
+        riot_ctrl(2, APP, Shell),
+        riot_ctrl(3, APP, Shell),
+    )
+    lladdrs = [dict(zip(("netif", "lladdr"), lladdr(node.ifconfig_list())))
+               for node in nodes]
+    from_addr = FROM_ADDR + "/64"
+    to_addr = TO_ADDR + "/64"
+    nodes[0].ifconfig_add(lladdrs[0]["netif"], from_addr)
+    nodes[-1].ifconfig_add(lladdrs[-1]["netif"], to_addr)
+    for i, node in enumerate(nodes):
+        if i < (len(nodes) - 1):
+            # set to-route
+            node.nib_route_add(lladdrs[i]["netif"], to_addr,
+                               lladdrs[i + 1]["lladdr"])
+        if i > 0:
+            # set from-route
+            node.nib_route_add(lladdrs[i]["netif"], from_addr,
+                               lladdrs[i - 1]["lladdr"])
+    return nodes
+
+
+@pytest.fixture
+def netif_parser():
+    return IfconfigListParser()
+
+
+# pylint: disable=W0621
+def _get_nodes_netifs(nodes, netif_parser):
+    nodes_netifs = []
+    for node in nodes:
+        netifs = netif_parser.parse(node.ifconfig_list())
+        key = next(iter(netifs))
+        nodes_netifs.append({
+            "netif": key,
+            "hwaddr": netifs[key]["long_hwaddr"],
+            "lladdr": [addr["addr"] for addr in netifs[key]["ipv6_addrs"] if
+                       addr["scope"] == "link"][0],
+        })
+    return nodes_netifs
+
+
+# pylint: disable=W0621
+def _l2filter_nodes(nodes, nodes_netifs):
+    for i, node in enumerate(nodes):
+        if i < (len(nodes) - 1):
+            # set to-route
+            node.ifconfig_l2filter_add(nodes_netifs[i]["netif"],
+                                       nodes_netifs[i + 1]["hwaddr"])
+        if i > 0:
+            # set from-route
+            node.ifconfig_l2filter_add(nodes_netifs[i]["netif"],
+                                       nodes_netifs[i - 1]["hwaddr"])
+
+
+# pylint: disable=W0621
```

Comment:
```suggestion
```

Reply:
Reasoning?

Reply:
Pylint didn't complain when ran locally? Did I do it wrong?

Reply:
Ah, you are right! `nodes_netifs` is used in those functions after all!